### PR TITLE
support for structs

### PR DESCRIPTION
**Pull request by afsalthaj**
Raising the PR again in an upstream branch to get rid of build permission issues, made changes as per the  review comments
https://github.com/CommBank/beeswax/pull/7

### CODE REVIEW

**Comment by afsalthaj** [comment]
@rowandavies  this is based on a previous PR that was raised quite a while ago. It took some time for me to understand that I need to push it to upstream branch to properly invoke the travis build. I have made some changes based a few of the review comments. Could you please look at it?

**Comment by afsalthaj** [comment]
@toddmowen thanks for the review. Il make sure that it works for list of structs. I will have to look at how we can have a separate behaviour for Scrooge and humbug. Can we have a design discussion? Should beeswax have a tight coupling with humbug (and its inherent restrictions)?

**Comment by afsalthaj** [comment]
@toddmowen I added support for list/map of structs and their tests. 
**On Separting out Humbug and Scrooge behaviour**
We can distinguish Humbug and Scrooge by checking if the `runTimeClass` contains an interface `HumbugThriftStruct`. I can do this. However, we should be aware that the whole project is currently developed and tested using `com.twitter.scrooge.ThriftStruct` for  the features common to humbug and Scrooge (asserting the exclusive features of Scrooge to exceptions). To prove this fact, we could remove the humbug plugins from the build, and all the tests still passed.

**Comment by afsalthaj** [comment]
@toddmowen  
Also, here are few key points that I consider to be the reasons for why we don't need explicit error messages for humbug thrifts:
1) A humbug thrift class extends com.twitter.scrooge.ThriftStruct with `HumbugThriftStruct`. Beeswax handles everything with just ` com.twitter.scrooge.ThriftStruct` and doesn't care about how ` _ <:ThriftStruct` is created.
2) As of now, it is impossible for anyone to pass a  `ThriftStruct` using humbug plugin to beeswax, as such a `ThriftStruct` can't exist.
3) If by any chance, if someone creates a complex thrift struct using humbug plugin in future, then it should be treated in the same way as it is now in beeswax (in this PR).

**Comment by toddmowen** [comment]
@afsalthaj In other words, beeswax works with all humbug-generated thrifts. Sure, that's no problem. I misunderstood, thinking you meant there would be some runtime or compile error if using humbug.

I'll finish the review today.

File: src/main/scala/au/com/cba/omnia/beeswax/HiveMetadataTable.scala [hunk 1]
```diff
@@ -135,15 +146,22 @@ object HiveMetadataTable {
       "bigint"
     else if (manifest[String] == mani)
       "string"
+    else if (mani.runtimeClass.getInterfaces.contains(classOf[ThriftStruct]))
+      structToHiveString(mani.runtimeClass)
+
```
Comment (toddmowen) [inline src/main/scala/au/com/cba/omnia/beeswax/HiveMetadataTable.scala]: Minor: seems unusual style to separate precede the "else" with a blank line.

**Pull request merged**